### PR TITLE
Adding docstring to coding guidelines

### PR DESCRIPTION
**Pull request by RazTamir**

### CODE REVIEW

File: docs/coding_guidelines.md [hunk 1]
```diff
@@ -74,6 +74,32 @@ print("This is old style of formatting and should be avoided:")
...
+* The description should be as follow:
+```python
+def my_example(arg1, arg2):
+    """
+    This function is for showing how a proper docstring should
+    look like
+
+    Args:
+        arg1 (<type of argument>): Description of what this argument is and if
+        the
```
Comment (clacroix12) [inline docs/coding_guidelines.md]: Should "the" be indented in 4 spaces with the rest following it?

Comment (RazTamir) [reply docs/coding_guidelines.md]: Thanks!

**Comment by shylesh** [comment]
LGTM

File: docs/coding_guidelines.md [hunk 2]
```diff
@@ -74,6 +74,31 @@ print("This is old style of formatting and should be avoided:")
...
+```python
+def my_example(arg1, arg2):
+    """
+    This function is for showing how a proper docstring should
+    look like
+
+    Args:
+        arg1 (<type of argument>): Description of what this argument is and if
+            the line is too long we will indent according to this (4 spaces)
+        arg2 (int): This argument is an integer
```
Comment (petr-balogh) [inline docs/coding_guidelines.md]: Do you think that we should also document default values?

I am used to do:
```python
def my_example(arg1, arg2=10):
    """
    This function is for showing how a proper docstring should
    look like

    Args:
        arg1 (<type of argument>): Description of what this argument is and if
            the line is too long we will indent according to this (4 spaces)
        arg2 (int): This argument is an integer (default: 10)
```

Comment (RazTamir) [reply docs/coding_guidelines.md]: Done

**Pull request merged**